### PR TITLE
fix: add index.action suffix for data entry redirect [DHIS2-6705]

### PR DESCRIPTION
**Pull request by jbee**
### Summary
This is a workaround so that the `/dhis-web-dataentry/` URL becomes `/dhis-web-dataentry/index.action` when doing the redirct 
after login. This is important so have the data entry work as start app.

### Manual Testing
* goto System settings => Appearance and select _Data Entry_ as _Start Page_
* logout
* login again, check the data entry app is displayed correctly as the first page the user sees

### CODE REVIEW

**Review by vietnguyen — APPROVED** [review]

File: dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/about/action/RedirectAction.java [hunk 1]
```diff
@@ -77,13 +77,21 @@ public String execute()
                     if ( app.getShortName().equals( startModule.substring( "app:".length() ) ) )
                     {
                         redirectUrl = app.getLaunchUrl();
+                        if ( redirectUrl.endsWith( "dhis-web-dataentry/" ) )
```
Comment (larshelge) [inline dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/about/action/RedirectAction.java:80]: Could we move `dhis-web-dataentry` to a final variable and re-use?

File: dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/about/action/RedirectAction.java [hunk 2]
```diff
@@ -77,13 +77,21 @@ public String execute()
...
+                        {
+                            redirectUrl += "index.action";
+                        }
```
Comment (larshelge) [inline dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/about/action/RedirectAction.java:83]: Just curious, if the `startModule` is a Struts module/the data entry module, does it start with `app:`?

Just wondering what the if / else logic is good for. I.e. if the data entry module starts with `app:`, then we don't need to put this special fix in both branches of the if / else.

```
if ( startModule.startsWith( "app:" ) )
```

Comment (jbee) [reply dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/about/action/RedirectAction.java:83]: I think it does not but I could not debug the class as I don*t have a working IDE config for the web modules.

Comment (larshelge) [reply dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/about/action/RedirectAction.java:83]: Okay, perhaps just a good old system.out.println?

Comment (jbee) [reply dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/about/action/RedirectAction.java]: removed it from that branch and tested it still works which it does.

File: dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/about/action/RedirectAction.java (hunk 1 above)
Comment (jbee) [reply dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/about/action/RedirectAction.java]: now there is only one occurence.

**Review by larshelge — APPROVED** [review]

**Pull request merged**